src/poly/commitment.rs

```diff
@@ -42,27 +42,38 @@ impl<C: CurveAffine> Params<C> {
 
         let n: u64 = 1 << k;
 
-        let g = {
-            let hasher = &T::init(io::sink(), C::Base::one());
+        let try_and_increment = |hasher: &Blake2bState| {
+            let mut trial = 0u64;
+            loop {
+                let mut hasher = hasher.clone();
+                hasher.update(&(trial.to_le_bytes())[..]);
+                let mut hash = [0u8; 32];
+                hash[..].copy_from_slice(hasher.finalize().as_bytes());
+                let p = C::from_bytes(&hash);
```

Comment:
This can probably be:
```suggestion
                let p = C::from_bytes(&hasher.finalize().as_bytes().try_into().unwrap());
```